File: core/utils/random.js
export function randomNum(min, max) {
    return (Math.random() * (max - min)) + min;
}

export function randomPick(arr) {
    return arr[Math.floor(randomNum(0, arr.length))]
}

export function randDiv(min, max, d) {
    const minK = Math.ceil(min / d);
    const maxK = Math.floor(max / d);

    const k = Math.floor(randomNum(minK, maxK + 1));
    return k * d;
}

File: void_front/game/core/utils/random.py
from __future__ import annotations

import random


def np.random.uniform(min_value, max_value):
    return (random.random() * (max_value - min_value)) + min_value


def randomPick(arr):
    return arr[int(np.random.uniform(0, len(arr)))]


def randDiv(min_value, max_value, d):
    from math import ceil, floor

    minK = ceil(min_value / d)
    maxK = floor(max_value / d)
    k = int(np.random.uniform(minK, maxK + 1))
    return k * d
